src/plugins/charset_converter.c

```diff
@@ -0,0 +1,185 @@
+#include "plugins/charset_converter.h"
+#include "flutter-pi.h"
+#include "pluginregistry.h"
+#include "util/logging.h"
+#include <iconv.h>
+
+static bool convert(char *buf, char *outbuf, size_t len, const char *from, const char *to)
+{
+    iconv_t iconv_cd;
+    if ((iconv_cd = iconv_open(to, from)) == (iconv_t) -1) {
+        LOG_ERROR("Cannot open iconv from %s to %s\n", from, to);
+        return false;
+    }
+
+    char *inbuf = buf;
+    size_t inlen = len;
+    size_t outlen = len;
+    size_t res = 0;
+
+    while (inlen > 0 && outlen > 0) {
+        res = iconv(iconv_cd, &inbuf, &inlen, &outbuf, &outlen);
+        if (res == 0)
+            break;
+
+        if (res == (size_t) (-1))
+        {
```

Comment:
Here and in other places, keep opening brackets on the same line:
```suggestion
        if (res == (size_t) (-1)) {
```

(There's also a clang format file which you can apply)